File: adventofcode2020/01/part1.py
nums: list[int] = []
with open("input.txt") as file:
    nums = [int(line) for line in file.readlines()]

for num1 in nums:
    for num2 in nums:
        if num1 + num2 == 2020:
            print(num1 * num2)
            exit(0)

File: adventofcode2023/01/part1.rb
sum = 0
File.readlines('input.txt').each do |line|
  first_num = nil
  last_num = nil
  line.each_char do |char|
    first_num = char if char =~ /[0-9]/ and first_num.nil?
    last_num = char if char =~ /[0-9]/
  end

  line_sum = (first_num + last_num).to_i
  sum += line_sum
end

puts sum
